src/pypromice/qc/github_data_issues.py

```diff
@@ -176,13 +176,11 @@ def adjustData(ds,
         # removing potential time shifts from the adjustment list
         adj_info = adj_info.loc[adj_info.adjust_function != "time_shift", :]
 
-        # if t1 is left empty, then adjustment is applied until the end of the file
-        adj_info.loc[adj_info.t0.isnull(), "t0"] = ds_out.time.values[0]
-        adj_info.loc[adj_info.t1.isnull(), "t1"] = ds_out.time.values[-1]
-        # making all timestamps timezone naive (compatibility with xarray)
-        adj_info.t0 = pd.to_datetime(adj_info.t0).dt.tz_localize(None)
-        adj_info.t1 = pd.to_datetime(adj_info.t1).dt.tz_localize(None)
-
+        # making sure that t0 and t1 columns are object dtype then replaceing nan with None
+        adj_info[['t0','t1']] = adj_info[['t0','t1']].astype(object)
```

Comment:
I don't understand why you cast the type to object. I suppose the dynamically inferred types are either
* `str` if all the values in the columns are strings
* `float` if all the values in the columns inferred as nan
* `object` if the values in the columns have different types

Reply:
I guess that, along with your next comment, it is a matter of personal preference:

I prefer replacing all the missing dates by `None` in two single-line vectorized call (l.181-182) rather than having `if`s in a `for` loop. But before I do that, I need to make sure that the t0 and t1 columns can accommodate `None`. If a t0 or t1 column has a `float` type and I try to replace the values by `None` pandas actually coerces those `None` into their float equivalent: `np.nan`. This causes problem later as `slice(np.nan, np.nan)` fails. When t0 and t1 are of object type, then replacing some of their values by `None` will keep the `None` as `NoneType`.